fix(ama-sdk): support of URL as spec-path input (#1572)
AmadeusITGroup/otter: packages/@ama-sdk/schematics/README.md

### How to use?

The typescript generator provides 3 generators:

- **shell**: To generate the "shell" of an SDK package
- **core**: To (re)generate the SDK based on a specified OpenApi specification
- **create**: To create a new SDK from scratch (i.e. chain **shell** and **core**)

> To generate the `shell` in an existing monorepo, you can run:

Reviewer: ```suggestion
You can generate the `shell` in an existing monorepo with the command:
```

> You can generate the `shell` in an existing monorepo with the command:

Reviewer 2: the command to generate in an existing monorepo is 
`yarn ng g sdk typescript sdkName`
or with schematics
`yarn schematics @o3r/workspace:sdk sdkName`

```shell
yarn schematics @ama-sdk/schematics:typescript-shell
```

or from scratch, with the NPM initializer:

```shell
npm create @ama-sdk typescript <project-name>
```

> The package generated comes with the following script in the package.json:

Reviewer 3: ```suggestion
The generated package comes with the following script in the package.json:
```

```json5
{
  // ...

>   "generate": "yarn schematics @ama-sdk/schematics:typescript-core --spec-path ./openapi.yaml",

Reviewer 3: ```suggestion
  "generate": "yarn schematics @ama-sdk/schematics:typescript-core --spec-path ./openapi-spec.yaml",
```

  "upgrade:repository": "yarn schematics @ama-sdk/schematics:typescript-shell"
}
```

> [!NOTE]

> > Use `generate` to (re)generate your SDK based on the content of `./swagger-spec.yaml` (make sure you have this file at the root of your project) and `upgrade:repository` to regenerate the structure of your project.

Reviewer 3: openapi.yaml instead of swagger-spec.yaml no? should be the same file name as in the generate command line above?

> [!TIP]

> > The `--spac-path` parameter does support YAML and JSON format file based on file system path or remote URL.

Reviewer: ```suggestion
> The `--spec-path` parameter supports YAML and JSON format file based on file system path or remote URL.
```

Reviewer 3: ```suggestion
> The `--spec-path` parameter supports YAML and JSON file formats based on the file system path or remote URL.
```

> If you use `Yarn2+` with PnP, you can modify the following `scripts` in `package.json` to generate the SDK based on specification in a dependency package:

Reviewer: ```suggestion
If you use `Yarn2+` with PnP, you can modify the following `scripts` in `package.json` to generate the SDK based on specifications in a dependency package:
```

### How to use?

The typescript generator provides 3 generators:

- **shell**: To generate the "shell" of an SDK package
- **core**: To (re)generate the SDK based on a specified OpenApi specification
- **create**: To create a new SDK from scratch (i.e. chain **shell** and **core**)

You can generate the `shell` in an existing monorepo with the command:

```shell
#Monorepo with Otter:
yarn ng g sdk sdkName

# Monorepo without Otter;
yarn schematics @o3r/workspace:sdk sdkName
```

or from scratch, with the NPM initializer:

```shell
npm create @ama-sdk typescript <project-name>
```

The generated package comes with the following script in the package.json:

```json5
{
  // ...

>   "generate": "yarn schematics @ama-sdk/schematics:typescript-core --spec-path ./openapi-spec.yaml",

Reviewer: We probably need to update it in the config we generate with the shell

      "example-sdk": { // any name you like (can be referenced using --generator-key)
        "generatorName": "typescriptFetch",
        "output": ".",

>         "inputSpec": "./swagger-spec.yaml" // support YAML and JSON format to file system path and remote URL

Reviewer 3: ```suggestion
        "inputSpec": "./swagger-spec.yaml" // supports YAML and JSON formats, based on the file system path or remote URL
```

Example:

```shell

> yarn schematics @ama-sdk/schematics:typescript-core --spec-path ./openapi.yaml --global-property debugModels,debugOperations

Reviewer 3: ./openapi-spec.yaml no? should be consistent throughout the documentation
